contracts/erc721/PaymentObligation.sol

```diff
@@ -228,7 +232,7 @@ contract PaymentObligation is Initializable, UserMintableERC721 {
       to,
       tokenId,
       anchorId,
-      merkleRoot_,
+      singatureRoot_,
```

Comment:
I think this is wrong. It should the `documentRoot`

Reply:
`    _tokenDetails[tokenId] = OwnedAnchor(anchorId, merkleRoot);`
I think we want to store the documentRoot

Reply:
Removed _tokenDetails from USerMintableERC721 and moved to PoDetails. We need to check all fields agains the signatureRoot.